rendering.php

```diff
@@ -849,6 +849,15 @@ public function renderNoteText($note) {
 
             $text .= ($publication != '') ? ',' : '.';
         }
+        elseif ($data->has('booktitle')) {  // rwduzhao
+            // $authors? $title //booktitle//, volume, $publication?
+
+            $text = $title . ' ' . $this->renderBookTitle($data);
+
+            // $authors? $text, $publication?
+
+            $text .= ($publication != '') ? ',' : '.';
```

Comment:
Refactor the code to avoid duplication. For example, create $subtitle variable, assign it to journal or booktitle, and use that variable to decide if it's a book or not.

```
        if ($data->has('journal')) {
            // $authors? $title //journal//, volume, $publication?
            $subtitle = ...
        }
        elseif ($data->has('booktitle')) {
            // $authors? $title //booktitle//, $publication?
            $subtitle = ...
        }

        // $authors? //$title// edition. $publication?
        // $authors? chapter In //$title// edition. $publication?
        // $authors? $title $subtitle?, $publication?

        if (!empty($subtitle)) {
            ...
```